### ISSUE TITLE
Stability of OptiminerZcash

### ISSUE THREAD

**Issue by afrogallego** [Optiminer/OptiminerZcash#11]
Hi, yesterday I tested your program, I liked it, but stopped working in an hour, error message, lost communication with http://zcash.flypool.org/. 

The miners are based on Sapphire RX 470 8GB, they work well, they have no problem.

I tested your program with the latest version of Xubuntu.

I would like to use OptiminerZcash, what can be the reason it is unstable? Greetings and thank you.

Sorry for my english, google

**Comment by Optiminer** [comment]
What error message did it give? Please provide the log file.

**Comment by grizscom** [comment]
Hello All
I join the request.
I like miners do stop (catch screeching halt) with the log:
-----
[2016-12-18 16:50:28.753] [info] [GPU1]  75.0 I/s 134.2 S/s (5s) 75.2 I/s 141.8 S/s (1m)
[2016-12-18 16:50:29.851] [warning] [GPU4] Share submission failed: "job not found"! (196 / 197)
[2016-12-18 16:50:29.923] [warning] [GPU1] Share submission failed: "job not found"! (196 / 198)
[2016-12-18 16:50:29.923] [warning] [GPU2] Share submission failed: "job not found"! (196 / 199)
[2016-12-18 16:50:29.947] [info] [GPU0]  101.0 I/s 186.6 S/s (5s) 100.9 I/s 189.4 S/s (1m)
[2016-12-18 16:50:29.995] [warning] [GPU3] Share submission failed: "job not found"! (196 / 200)
[2016-12-18 16:50:29.995] [warning] [GPU3] Share submission failed: "job not found"! (196 / 201)
[2016-12-18 16:50:30.059] [info] [GPU2] Share accepted! (197 / 202)
[2016-12-18 16:50:30.059] [info] [GPU0] Share accepted! (199 / 204)
[2016-12-18 16:50:30.059] [info] [GPU4] Share accepted! (198 / 203)
[2016-12-18 16:50:30.117] [info] [GPU3] Share accepted! (200 / 205)
[2016-12-18 16:50:32.525] [info] [GPU2]  74.0 I/s 138.6 S/s (5s) 74.8 I/s 140.1 S/s (1m)
[2016-12-18 16:50:32.549] [info] [GPU4]  99.0 I/s 184.6 S/s (5s) 98.2 I/s 182.7 S/s (1m)
[2016-12-18 16:50:32.580] [info] [GPU3]  89.0 I/s 169.2 S/s (5s) 89.0 I/s 165.3 S/s (1m)
[2016-12-18 16:50:33.027] [error] IO/Error communicating to zec-eu.coinmine.pl
[2016-12-18 16:50:33.027] [error] IO/Error communicating to zec-eu.coinmine.pl
[2016-12-18 16:50:33.027] [error] IO/Error communicating to zec-eu.coinmine.pl
[2016-12-18 16:50:33.048] [info] Connecting to zec-eu.coinmine.pl:8008.
[2016-12-18 16:50:33.053] [info] GPU3 Waiting for work from pool.
[2016-12-18 16:50:33.054] [info] GPU1 Waiting for work from pool.
[2016-12-18 16:50:33.057] [info] GPU0 Waiting for work from pool.
[2016-12-18 16:50:33.074] [info] GPU4 Waiting for work from pool.
[2016-12-18 16:50:33.090] [info] GPU2 Waiting for work from pool.
-----
Please do something to reconnect to the pool.

**Comment by Optiminer** [comment]
@grizscom From your log you can see that it tried to reconnect to zec-eu.coinmine.pl:8008
[2016-12-18 16:50:33.048] [info] Connecting to zec-eu.coinmine.pl:8008.
Are you sure that the pool was up and reachable from your computer at this time? The miner will generally keep trying to reconnect until it succeeds.

**Comment by grizscom** [comment]
@Optiminer 
Yes, I have about 30 rigs at one place, rest others was online and mining at that time.
Same thing happened now, there are logs:
http://pastebin.com/8QhnXUv4
http://pastebin.com/LfBMUNxG

**Comment by Optiminer** [comment]
Thanks for the logs. I will try to reproduce this on coinmine.pl. Meanwhile you could kill the miner in the watchdog script which should help in this case as a workaround.

**Comment by grizscom** [comment]
@Optiminer 
Thanks for your software, but but this bug brought me half a day of downtime.
Unfortunately, I am migrating to another miner.

**Comment by rbiegel** [comment]
optiminer seems to give up reconnection attempts here after 20 seconds.

```
[2017-01-09 23:38:47.680] [info] Optiminer/Zcash 1.3.1 (C) Optiminer 2017
...
[2017-01-09 23:56:02.815] [info] [Total]  319.0 I/s 598.8 S/s (5s) 318.7 I/s 594.4 S/s (1m)
[2017-01-09 23:56:03.456] [info] [GPU0]  107.0 I/s 200.8 S/s (5s) 106.2 I/s 196.5 S/s (1m)
[2017-01-09 23:56:04.463] [info] Concurrent write detected, queuing.
[2017-01-09 23:56:05.513] [info] [GPU2]  106.0 I/s 191.6 S/s (5s) 106.2 I/s 196.9 S/s (1m)
[2017-01-09 23:56:06.745] [info] [GPU1]  106.0 I/s 192.2 S/s (5s) 106.2 I/s 199.0 S/s (1m)
[2017-01-09 23:56:07.455] [warning] The write deadline expired! Closing connection.
[2017-01-09 23:56:07.455] [error] IO/Error communicating to eu1-zcash.flypool.org
```
last message repeated 25 times within a timespan of 1 ms.
```
[2017-01-09 23:56:07.472] [info] GPU1 Waiting for work from pool.
[2017-01-09 23:56:07.476] [info] Connecting to eu1-zcash.flypool.org:3333.
[2017-01-09 23:56:07.519] [info] GPU2 Waiting for work from pool.
[2017-01-09 23:56:07.521] [info] GPU0 Waiting for work from pool.
[2017-01-09 23:56:27.497] [error] No such host: eu1-zcash.flypool.org(Host not found (non-authoritative), try again later)
```
The log ends with these lines. Most of the time the optiminer process is still running (0% CPU usage) but this isn't always the case.

The connection problem might be caused by the ISP forcing a reconnection after a specific uptime but I think optiminer should be able to recover without using the watchdog functionality.

**Comment by Optiminer** [comment]
This is fixed in 1.3.2.
